### ISSUE TITLE
fails to parse entries with no PID

### ISSUE THREAD

**Issue by mhmeadows63** [vdemedes/syslog-parse#3]
Example: `<38>Jul 18 08:29:33 portal systemd-logind: Removed session c160.`

This suffers two issues:
1. the process-name contains a dash
2. the process-id is absent

I recommend tweaking the regular expression as follows:
`/(\<[0-9]+\>)?([a-z]{3})\s+([0-9]{1,2})\s+([0-9]{2})\:([0-9]{2})\:([0-9]{2})(\s+[a-z0-9\.\-]+)?\s+([a-z0-9\-]+)(?:\[([a-z0-9\-\.]+)\])?\:(.+)/i`

Disection (with notes):

```
/(\<[0-9]+\>)?                      [code]
([a-z]{3})\s+([0-9]{1,2})\s+        [month, day]
([0-9]{2})\:([0-9]{2})\:([0-9]{2})  [hour, minute, second]
(\s+[a-z0-9\.\-]+)?                 [hostname]   (also allow digits in hostname)
\s+([a-z0-9\-]+)                    [procname]  (also allow dashes in procname)
(?:\[([a-z0-9\-\.]+)\])?\:          [pid]   (now optional)
(.+)/i                              [message]
```

No other changes are necessary.

**Comment by vadimdemedes** [comment]
Would you be interested in making a PR to fix this?

**Comment by mhmeadows63** [comment]
Yes, but being a git newbie, I am clueless how to do it.
Can you point me at any how-to for guidance?
- Martin (Galaxy S4)  -------------------

-------- Original message --------

**Comment by vadimdemedes** [comment]
No worries, https://www.sammyk.me/how-to-contribute-to-an-open-source-project-on-github seems like a good guide to start with.

**Comment by vadimdemedes** [comment]
Forgot to ask, is missing PID aligns with a syslog specification?

**Comment by mhmeadows63** [comment]
I'm not certain, but api calls can optionally enable it, so I assume so.
e.g. https://www.npmjs.com/package/modern-syslog#syslogoption
I am working on centos7/pi3b and the original example was from the base configuration.
- Martin (Galaxy S4)  -------------------

**Comment by mhmeadows63** [comment]
I hope this is what you wanted

**Comment by vadimdemedes** [comment]
Fixed by #4.
